vectors/no_resizable_vec.rs

```diff
@@ -111,233 +96,189 @@ impl<T> NoResizableVec<T> {
             None
         } else {
             self.len -= 1;
-            unsafe { Some(ptr::read(self.ptr().add(self.len))) }
-        }
-    }
-
-    pub fn find(& self, elem: &T) -> Option<usize>
-        where  T: Ord, {
-        let mut i = 0;
-        while i < self.len {
-            if self.index(i).cmp(elem) ==  Ordering::Equal {
-                return Some(i)
+            unsafe {
+                let end: *mut T = self.buf.ptr.as_ptr().add(self.len);
+                Some(end.read())
             }
-            i += 1;
         }
-        return None
     }
 
-    pub fn clone(&self) -> Self {
-        let vec = RawVec::new(self.cap());
-        let res = NoResizableVec {
-            buf: vec,
-            len: self.len,
-        };
-        unsafe {
-            ptr::copy_nonoverlapping(
-                self.ptr(),
-                res.ptr() ,
-                self.len
-            )
-        }
-        res
-    }
-
-    pub fn insert(&mut self, index: usize, elem: T) {
+    pub fn insert(&mut self, index: usize, value: T) {
+        assert!(self.buf.cap > self.len);
         assert!(index < self.len);
-        assert!(self.len < self.cap());
         unsafe {
-            ptr::copy(
-                self.ptr().add(index),
-                self.ptr().add(index + 1),
-                self.len - index,
-            );
-            ptr::write(self.ptr().add(index), elem);
-            self.len += 1;
+            let ptr: *mut T = self.buf.ptr.as_ptr().add(index);
+            ptr.copy_to(ptr.add(1), self.len - index);
+            ptr.write(value);
         }
+        self.len += 1;
     }
 
     pub fn remove(&mut self, index: usize) -> T {
         assert!(index < self.len);
         unsafe {
             self.len -= 1;
-            let result = ptr::read(self.ptr().add(index));
-            ptr::copy(
-                self.ptr().add(index + 1),
-                self.ptr().add(index),
-                self.len - index,
-            );
-            result
+            let ptr: *mut T = self.buf.ptr.as_ptr().add(index);
+            let value: T = ptr.read();
+            ptr.add(1).copy_to(ptr, self.len - index);
+            value
         }
     }
 
-    pub fn drain(&mut self, range: Range<usize>) -> Drain<T> {
-        unsafe {
-            let slice: &[T] = &*self;
-            let slice: &[T] = &slice[range.clone()];
-            let iter: RawValIter<T> = RawValIter::new(slice);
-
-            Drain {
-                const_range: range.clone(),
-                iter: iter,
-                // vec: PhantomData,
-                vec: self,
+    pub fn find(&self, value: &T) -> Option<usize> where T: Ord {
```

Comment:
what happens when `value` is not found? There is no return, is that ok?

Reply:
What do you mean? `None` is returned at the end on line 137

Reply:
Sorry, I got lost in the diff. Thanks for pointing this.